app/controllers/protocols_controller.rb

```diff
@@ -860,7 +860,8 @@ def load_organization_and_type
   def check_view_all_permissions
     load_organization_and_type
 
-    unless can_view_organization_protocols(@current_organization)
+    if @current_organization &&
+       !can_view_organization_protocols(@current_organization)
```

Comment:
Isn't it enough to check presence of @current_organization variable in the view to enable/disable protocols button? Because even if variable present but contains wrong data it will be filtered out later in protocols controller. And by simpler checks in the view we'll reduce load as it is navbar and it's present in almost every view.